includes/object-cache.php

```diff
@@ -636,6 +640,116 @@ protected function connectUsingPredis( $parameters ) {
         );
     }
 
+    /**
+     * Connect to Redis using the Credis library.
+     *
+     * @param  array $parameters
+     * @return void
+     */
+    protected function connectUsingCredis( $parameters ) {
+        $client = 'Credis';
+
+        // Require PHP 5.4 or greater
+        if ( version_compare( PHP_VERSION, '5.4.0', '<' ) ) {
+            throw new Exception( 'Predis requires PHP 5.4 or newer.' );
+        }
+
+        $creds_path = sprintf(
+            '%s/redis-cache/dependencies/vendor/colinmollenhour/credis/',
+            defined( 'WP_PLUGIN_DIR' ) ? WP_PLUGIN_DIR : WP_CONTENT_DIR . '/plugins'
+        );
+
+        $to_load = [];
+
+        // Load bundled Credis library.
+        if ( ! class_exists( 'Credis_Client' ) ) {
+            $to_load[] = 'Client.php';
+        }
+
+        if ( ( defined( 'WP_REDIS_SHARDS' ) || defined( 'WP_REDIS_SENTINEL' ) || defined( 'WP_REDIS_SERVERS' ) || defined( 'WP_REDIS_CLUSTER' ) ) && ! class_exists( 'Credis_Cluster' ) ) {
+            $to_load[] = 'Cluster.php';
+
+            if ( true
```

Comment:
Debug?

Reply:
Actually I do write all conditions with more than one condition using this approach as it allows for fast order switching or commenting out. Also I personally am a line 80 guy :wink:. The compiler removes the first boolean anyway. 
Forgot to transform it back, will update in a couple if hours.

Reply:
changed in https://github.com/rhubarbgroup/redis-cache/pull/189/commits/f03b1572e3f78679ab27b062b5cde40ba4d2a9e5